src/python/tests/core/bot/fuzzers/libFuzzer/engine_test.py

```diff
@@ -59,11 +71,176 @@ def test_prepare(self):
-    self.assertItemsEqual(['/new_corpus_dir'], options.additional_corpus_dirs)
+    self.assertItemsEqual(['/new_corpus_dir', '/corpus_dir'],
+                          options.fuzz_corpus_dirs)
     self.assertDictEqual({'extra_env': '1'}, options.extra_env)
     self.assertFalse(options.use_dataflow_tracing)
     self.assertTrue(options.is_mutations_run)
+
+    self.mock.unpack_seed_corpus_if_needed.assert_called_with(
+        '/path/target', '/corpus_dir')
+
+  def test_prepare_invalid_dict(self):
+    """Test prepare with an invalid dict path."""
+    with open('/path/target.options', 'w') as f:
+      f.write('[libfuzzer]\n'
+              'max_len=31337\n'
+              'timeout=11\n'
+              'dict=nope.dict\n')
+
+    engine_impl = engine.LibFuzzerEngine()
+    options = engine_impl.prepare('/corpus_dir', '/path/target', '/path')
+    self.assertItemsEqual(
+        ['-max_len=31337', '-timeout=11', '-rss_limit_mb=2048', '-arg1'],
+        options.arguments)
+
+  def test_prepare_auto_add_dict(self):
+    """Test prepare automatically adding dict argument."""
+    with open('/path/target.options', 'w') as f:
+      f.write('[libfuzzer]\n' 'max_len=31337\n' 'timeout=11\n')
+    self.fs.create_file('/path/target.dict')
+
+    engine_impl = engine.LibFuzzerEngine()
+    options = engine_impl.prepare('/corpus_dir', '/path/target', '/path')
+    self.assertItemsEqual([
+        '-max_len=31337', '-timeout=11', '-rss_limit_mb=2048', '-arg1',
+        '-dict=/path/target.dict'
+    ], options.arguments)
+
+
+class FuzzTest(fake_fs_unittest.TestCase):
+  """Fuzz() tests."""
+
+  def setUp(self):
+    # Set up fake filesystem.
+    test_helpers.patch_environ(self)
+    test_utils.set_up_pyfakefs(self)
+
+    self.fs.create_dir('/corpus')
+    self.fs.create_dir('/fuzz-inputs')
+    self.fs.create_dir('/fake')
+    self.fs.create_file('/target')
+    self.fs.add_real_directory(TEST_DIR)
+
+    test_helpers.patch(self, [
+        'bot.fuzzers.libfuzzer.LibFuzzerRunner.fuzz',
+        'bot.fuzzers.libfuzzer.LibFuzzerRunner.merge',
+        'os.getpid',
+    ])
+
+    os.environ['JOB_NAME'] = 'libfuzzer_asan_job'
+    os.environ['FUZZ_INPUTS_DISK'] = '/fuzz-inputs'
+
+    self.mock.getpid.return_value = 9001
+    self.maxDiff = None  # pylint: disable=invalid-name
+
+  def test_fuzz(self):
+    """Test fuzz."""
+    engine_impl = engine.LibFuzzerEngine()
+    options = engine.LibFuzzerOptions(
+        '/corpus',
+        ['-arg=1', '-timeout=123', '-dict=blah.dict', '-max_len=9001'], [],
+        ['/corpus'], {}, False, False)
+
+    with open(os.path.join(TEST_DIR, 'crash.txt')) as f:
+      fuzz_output = f.read()
+
+    def mock_fuzz(*args, **kwargs):  # pylint: disable=unused-argument
+      """Mock fuzz."""
+      self.fs.create_file('/fuzz-inputs/temp-9001/new/A')
+      self.fs.create_file('/fuzz-inputs/temp-9001/new/B')
+      return new_process.ProcessResult('command', 0, fuzz_output, 2.0, False)
```

Comment:
nit: named argument can help to make things readable. noticed this in a bunch of places, like where bools are used. Can help with future readability and maintenance.

Reply:
made these named arguments.